File: service_examples/src/class_service_server.py
#!/usr/bin/env python3

import rospy
from service_examples.srv import custom_srv,custom_srvRequest,custom_srvResponse


class Server:
    def __init__(self):
        self.server = rospy.Service('speed_check',custom_srv,self.cli)
    def cli(self,msg=custom_srvRequest()):
        result = custom_srvResponse()
        if msg.srvB.linear.x >0:
            rospy.loginfo('ok')
            result.srvC.data = True
        else :
            result.srvC.data = False

        return result


def main():
    S = Server()
    rospy.spin()

    


if __name__ == '__main__':
    rospy.init_node('class_service_server')
    try :
        main()
    except rospy.ROSException as e:
        print(e)

File: service_examples/src/class_service_server.cpp
#include<ros/ros.h>
#include<service_examples/custom_srv.h>
#include<iostream>

using namespace std;


class SerServer
{
    ros::NodeHandle n;
    ros::ServiceServer server;
    service_examples::custom_srv ji;
    public:
        SerServer();    
        bool Callback(service_examples::custom_srvRequest &req ,
                         service_examples::custom_srvResponse &res);
};

SerServer::SerServer()
{
    server = n.advertiseService("speed_check",&SerServer::Callback,this);

}

bool SerServer::Callback(service_examples::custom_srvRequest &req ,
                         service_examples::custom_srvResponse &res)
{
    if (req.srvB.linear.x > 0)
    {
        res.srvC.data = true;
        cout<<"ok\n";
    }
    else 
    {
        res.srvC.data = false;
    }
    return true;
}


int main(int argc,char ** argv)
{
    ros::init(argc,argv,"class_service_server");
    SerServer serve;
    ros::spin();
    return 0;   

}